File: articles/institutional-memory-knowledge-retention-and-organizational-continuity/go/institutional_memory_score.go
package main

import "fmt"

type InstitutionalMemoryCase struct {
	DocumentedRetention    float64
	TacitTransfer          float64
	Accessibility          float64
	InterpretiveUse        float64
	Revisability           float64
	TechnicalContinuity    float64
	MetadataQuality        float64
	DistributedIntegration float64
	MemoryJustice          float64
	PathDependencePressure float64
	LossFragmentation      float64
	SelectiveNarration     float64
	TurnoverPressure       float64
	KeyPersonDependency    float64
}

func InstitutionalMemoryScoreRaw(x InstitutionalMemoryCase) float64 {
	return 0.12*x.DocumentedRetention +
		0.12*x.TacitTransfer +
		0.12*x.Accessibility +
		0.12*x.InterpretiveUse +
		0.11*x.Revisability +
		0.09*x.TechnicalContinuity +
		0.08*x.MetadataQuality +
		0.08*x.DistributedIntegration +
		0.08*x.MemoryJustice -
		0.11*x.PathDependencePressure -
		0.11*x.LossFragmentation -
		0.08*x.SelectiveNarration -
		0.07*x.TurnoverPressure -
		0.06*x.KeyPersonDependency
}

func main() {
	demo := InstitutionalMemoryCase{84, 78, 80, 76, 74, 79, 82, 72, 70, 24, 22, 20, 26, 28}
	fmt.Printf("Institutional memory raw score: %.2f\n", InstitutionalMemoryScoreRaw(demo))
}

File: articles/institutional-memory-knowledge-retention-and-organizational-continuity/cpp/institutional_memory_score.cpp
#include <iostream>

struct InstitutionalMemoryCase {
    double documented_retention;
    double tacit_transfer;
    double accessibility;
    double interpretive_use;
    double revisability;
    double technical_continuity;
    double metadata_quality;
    double distributed_integration;
    double memory_justice;
    double path_dependence_pressure;
    double loss_fragmentation;
    double selective_narration;
    double turnover_pressure;
    double key_person_dependency;
};

double institutional_memory_score_raw(const InstitutionalMemoryCase& x) {
    return
        0.12 * x.documented_retention +
        0.12 * x.tacit_transfer +
        0.12 * x.accessibility +
        0.12 * x.interpretive_use +
        0.11 * x.revisability +
        0.09 * x.technical_continuity +
        0.08 * x.metadata_quality +
        0.08 * x.distributed_integration +
        0.08 * x.memory_justice -
        0.11 * x.path_dependence_pressure -
        0.11 * x.loss_fragmentation -
        0.08 * x.selective_narration -
        0.07 * x.turnover_pressure -
        0.06 * x.key_person_dependency;
}

int main() {
    InstitutionalMemoryCase demo {84, 78, 80, 76, 74, 79, 82, 72, 70, 24, 22, 20, 26, 28};
    std::cout << "Institutional memory raw score: "
              << institutional_memory_score_raw(demo) << std::endl;
    return 0;
}
